### ISSUE TITLE
Postgres dialect ALTER TABLE ADD CONSTRAINT support

### ISSUE THREAD

**Issue by psukys** [sqlfluff/sqlfluff#4257]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

sqlfluff complains  about unparsable section for:

```
ALTER TABLE schema1.table_1
  ADD CONSTRAINT my_constraint FOREIGN_KEY(table_3_fk)
  REFERENCES schema2.table_3(table_3_pk);
```

### Use case

From [https://www.postgresql.org/docs/current/sql-altertable.html](https://www.postgresql.org/docs/current/sql-altertable.html) , there's:
```
...
    ADD table_constraint [ NOT VALID ]
    ADD table_constraint_using_index
...
```
which can build up to aforementioned SQL statement

### Dialect

postgres

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by greg-finley** [comment]
Always feel free to add any support, even if not exhaustive. It will help us at least address some cases and it can be extended later.

**Comment by AidanHarveyNelson** [comment]
Hi This is parsing incorrectly because FOREIGN KEY is specified incorrectly.

The SQL Code you have mentioned above is considered invalid syntax in Postgres.
```
ALTER TABLE schema1.table_1 ADD CONSTRAINT my_constraint FOREIGN_KEY(table_3_fk) REFERENCES schema2.table_3(table_3_pk);
```

![image](https://user-images.githubusercontent.com/79121034/217098667-737c232e-d313-4d4f-8e6d-081d35ae99e5.png)

The below Syntax is considered valid, this also aligns with what is specified in the documentation.

```
ALTER TABLE schema1.table_1 ADD CONSTRAINT my_constraint FOREIGN KEY(table_3_fk) REFERENCES schema2.table_3(table_3_pk);
```

![image](https://user-images.githubusercontent.com/79121034/217099353-0fe9f258-b166-4bea-9fb1-c23acb38c16c.png)


This syntax is also parsable in SQL Fluff.

If you believe the above information is incorrect could you please provide evidence of the initial syntax provided being valid in Postgres (Screenshot of working command would be fine).

**Comment by psukys** [comment]
> If you believe the above information is incorrect could you please provide evidence of the initial syntax provided being valid in Postgres (Screenshot of working command would be fine).

Indeed, sorry for not resolving the issue before, this was a syntax error as metnioned

**Comment by mathieulb** [comment]
With psql 14.7 here, the above ALTER TABLE without schema gives only «relation "table_1" does not exist», not a syntax error. Also, the current manual says name is «The name (optionally schema-qualified) of an existing table to alter», and so do the manual of all past supported versions and beyond. I don't know why you have a different experience. https://www.postgresql.org/docs/current/sql-altertable.html

When I tried psql & pgadmin in the past, there was no default schema, but this only meant the tables were not found, no actual syntax errors, and there is a way to make tables accessible without explicit schema : `SET search_path TO` https://www.postgresql.org/docs/current/ddl-schemas.html

But the reason I am on this page is that I got the same parse error in MySQL, and was wondering whether the same grammatical rule could be copied to the MySQL module... only to find that it hasn't been added to the Postgres module either.
